Sources/SnapshotTesting/Strategy/View.swift

```diff
@@ -62,34 +62,58 @@ extension NSView: DefaultSnapshottable {
 #elseif os(iOS) || os(tvOS)
 extension Strategy where Snapshottable == UIView, Format == UIImage {
   public static var image: Strategy {
-    return .image(precision: 1, size: nil)
+    return .image()
   }
 
-  public static func image(precision: Float) -> Strategy {
-    return .image(precision: precision, size: nil)
-  }
+  public static func image(
+    drawingHierarchyInKeyWindow: Bool = false,
+    precision: Float = 1,
+    size: CGSize? = nil,
+    traits: UITraitCollection = .init()
+    )
+    -> Strategy {
 
-  public static func image(precision: Float = 1, size: CGSize) -> Strategy {
-    return .image(precision: precision, size: .some(size))
-  }
+      return SimpleStrategy.image(precision: precision).asyncPullback { view in
+        let size = size ?? view.frame.size
+        guard size.width > 0, size.height > 0 else {
+          fatalError("View not renderable to image at size \(view.frame.size)")
+        }
+        view.frame.size = size
 
-  static func image(precision: Float, size: CGSize?) -> Strategy {
-    return SimpleStrategy.image(precision: precision).asyncPullback { view in
-      let initialSize = view.frame.size
-      if let size = size { view.frame.size = size }
-      guard view.frame.width > 0, view.frame.height > 0 else {
-        fatalError("View not renderable to image at size \(view.frame.size)")
-      }
-      return view.snapshot ?? Async { callback in
-        addImagesForRenderedViews(view).sequence().run { views in
-          Strategy<CALayer, UIImage>.image.snapshotToDiffable(view.layer).run { image in
-            callback(image)
-            views.forEach { $0.removeFromSuperview() }
-            view.frame.size = initialSize
+        let child = UIViewController()
+        child.view.bounds = view.bounds
+        child.view.addSubview(view)
+        let parent = traitController(for: child, size: size, traits: traits)
+
+        let initialFrame = parent.view.frame
+        if drawingHierarchyInKeyWindow, let window = UIApplication.shared.keyWindow, window != view {
+          parent.view.frame.origin = CGPoint(x: .max, y: .max)
+          window.addSubview(parent.view)
+        }
+        parent.view.setNeedsLayout()
+        parent.view.layoutIfNeeded()
+        return parent.view.snapshot ?? Async { callback in
+          addImagesForRenderedViews(parent.view).sequence().run { views in
+            let cleanup = {
+              views.forEach { $0.removeFromSuperview() }
+              view.frame = initialFrame
```

Comment:
can this be a `defer`? not sure I'm fully following this code correctly.

Reply:
unfortunately not since one of the paths happens in a callback :(